src/Zir.zig

```diff
@@ -115,18 +116,13 @@ pub fn extraData(code: Zir, comptime T: type, index: usize) ExtraData(T) {
     };
 }
 
-/// TODO migrate to use this for type safety
 pub const NullTerminatedString = enum(u32) {
+    none = 0,
```

Comment:
there's a bug somewhere that prevents `none` from being non-zero. (sorry, i can't investigate it as i don't have llvm built locally.)

Reply:
I think this should be `empty` instead of `none`. [There will always be a 0 byte at index 0](https://github.com/ziglang/zig/blob/a09ba455c2b8334f790fc3ebca9a12c8bd06f1db/src/AstGen.zig#L134-L135), making it `""`. Also from the doc  comments of StructDecl, we see:

```zig
    ///        name: u32, // null terminated string index
    ///        - 0 means comptime or usingnamespace decl.
    ///          - if name == 0 `is_exported` determines which one: 0=comptime,1=usingnamespace
    ///        - 1 means test decl with no name.
    ///        - 2 means that the test is a decltest, doc_comment gives the name of the identifier
```

so, it should be:

```zig
pub const NullTerminatedString = enum(u32) {
    empty = 0,
    unnamed_test_decl = 1,
    decltest = 2,
    _,
};
```

If `none` is needed then a new type `OptionalNullTerminatedString` should be added.